src/mpyl/project.py

```diff
@@ -111,6 +111,60 @@ def from_config(values: dict):
         )
 
 
+@dataclass(frozen=True)
+class SecretKeyRef:
+    name: str
+    key: str
+    optional: bool
+    key_name: str = "secretKeyRef"
+
+    @staticmethod
+    def from_config(values: dict):
+        return SecretKeyRef(
+            name=values.get("name"),
+            key=values.get("key"),
+            optional=values.get("optional", False),
```

Comment:
is `optional` optional? 😅

Reply:
No it's actually not :D

Reply:
which checks?

Reply:
if not key or not name or not optional triggers if optional is False even though from a typing perspective it is a Some(False) and not a None